### PR TITLE
OOP Project: Milestone - 3: Game Logic

### PR DESCRIPTION
**Pull request by YuvrajM108**
Highlights on this milestone:
- Create a Ruby class for the game.
- Add methods to the Ruby class for designing the game logic.
- Make use of the Ruby class and its methods in the main file to implement game logic.

### CODE REVIEW

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight
 - [ ] You have mastered Git flow, the folder structure is organized, and the repo is professional.
- [ ] Code follows the DRY rule which is a good sign that it's clean and efficient.
- [ ] Game workes perfectly and as expected.
## But there are some changes to make in order to get this milestone approved 🙏
- [x] Game still shows a draw if player1 wins on the very last move. KIndly fix that!
- [x] The project should have at least two classes. It can be three as Game, Player, and Board. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/classes.png).
### Happy Coding ☺️

## If you have any questions you can comment here in this pull request or can always connect with me on slack ( @Sajjad Ahmad ).

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+class GameState
+  attr_reader :game_board, :game_state
+
+  def initialize()
+    # 9 Board Cells
```
Comment (Dipeshtwis) [inline lib/game.rb:5]: Remove the comment.

File: lib/game.rb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    checker = false
+    checker = true if compare_cells?(@game_board[0], @game_board[1], @game_board[2])
+    checker = true if compare_cells?(@game_board[3], @game_board[4], @game_board[5])
+    checker = true if compare_cells?(@game_board[6], @game_board[7], @game_board[8])
+    checker
+  end
+
+  def game_over?(_cell_number = nil)
+    game_over = false
+    # Full board
```
Comment (Dipeshtwis) [inline lib/game.rb:47]: Here as well

**Review by Dipeshtwis — APPROVED** [review]
# Status: Approved ✔️

## Hi @YuvrajM108 &  @omar-labana

Good job on implementing the suggested changes

- [ ] Please remove all the comments from the game.rb.

**You may close this Pull Request and merge the changes**
  
**Keep being awesome devs and good luck with your next milestone!**
---------

Cheers and Happy coding!👏👏👏

Code review by **Dipesh Kumar**
(Feel free to contact me on slack by the handle of  **deep** if you have any additional questions).
[Dipesh Kumar](https://deeptwist.tk) | [Twitter](https://twitter.com/97deepeshkumar)

**Pull request merged**